### ISSUE TITLE
[Request] macOS Standalone Bundle

### ISSUE THREAD

**Issue by mattbox** [chatty/chatty#171]
It would be nice to get a standalone mac app build of Chatty and wouldn't take too much effort be since all you really just need the jar files and a property list.
I tested out some options in the build file for standalone mac app using a Gradle plugin. https://github.com/mattbox/chatty/blob/master/build.gradle It's not pull request ready, but works pretty well. 
There's some other tweaks that could be added to make Chatty more mac friendly, but really just an executable would be make Chatty a lot easier to work with on mac.

**Comment by tduva** [comment]
There already is something like this: https://github.com/dehesselle/chatty

I can't test these things though, since I don't have a Mac.

**Comment by mattbox** [comment]
Yeah, I saw that fork, it's nice however it uses a separate build script which only works on mac. The plugin I'm using will build on any OS.

I see, are you the only one check pull requests? There aren't any other contributors that can do testing?

**Comment by dehesselle** [comment]
I started my fork because I couldn't find any existing solution for macOS. I wouldn't mind retiring it in favor of a multi-platform solution that could eventually end up as PR.

I'm not familiar with Gradle and I don't speak Java, but if testing efforts are limited to "check if the build is successfull and produces a working Chatty", I'd be willing to help.

**Comment by tduva** [comment]
I'm the only maintainer of Chatty, but if others that have access to a Mac can help test, I'd be willing to consider building a Mac version (that I can build on Windows) as well.

**Comment by mattbox** [comment]
I'd be willing to help with testing as well. There isn't anything special to the build that requires it be done on a mac, it's just the matter of using a cross-platform build script. The only issue I've run into so far is that you need a copy of the Mac version of the JRE if you want to build a standalone App on windows. 
It works, but I'm not too savvy with Java or Gradle there's probably a better way to make it work.

**Comment by bap14** [comment]
I've been using this for a while now on my mac to bundle the java up into a DMG and run it like a native app.  It's a simple bash script.  The only requirement is jq, but that's only because I'm reading the GitHub API data:

```
#!/bin/bash

json=$(curl -s "https://api.github.com/repos/chatty/chatty/releases/latest")

downloadURL=$(echo $json | jq '.assets[0].browser_download_url' | sed -e 's/^"//' -e 's/"$//')
appVersion=$(echo $json | jq '.tag_name' | sed -e 's/^"v//' -e 's/"$//')

echo "Building new Chatty release ${appVersion}"

cd ~/Downloads
curl -Lo Chatty_${appVersion}.zip "${downloadURL}"
unzip -d Chatty_${appVersion} Chatty_${appVersion}.zip
cd -

jdk=$(/usr/libexec/java_home)
$jdk/bin/javapackager -deploy \
	-v \
	-native dmg \
	-appclass chatty.Chatty \
	-name Chatty \
	-title "Chatty - The Twitch Chat Client" \
	-outdir deploy \
	-outfile Chatty \
	-BappVersion="${appVersion}" \
	-Bicon=package/macosx/dmgIcon.icns \
	-srcdir /Users/bap14/Downloads/Chatty_${appVersion} \
	-srcfiles Chatty.jar -srcfiles sounds -srcfiles img
```

The "package/macosx/dmgIcon.icns" is just the converted chatty icon: [dmgIcon.icns.zip](https://github.com/chatty/chatty/files/2094527/dmgIcon.icns.zip).

**Comment by dehesselle** [comment]
A lot of time has passed since I last visited here. - I just brought my fork up to date. It's still completely standalone / relocatable, signed (new!), brings it's own Java and Python 3 (new!) to power Streamlink and can use IINA now as alternative option to the Streamlink+VLC duo.

> consider building a Mac version (that I can build on Windows) as well

This will never be possible (or at least not in a satisfactory way) due to the peculiarities of the macOS platform (e.g. signing and soon notarizing which can only be done on a Mac). Having someone else maintain and contribute this part is out of the question?

**Comment by AnAnonymousUsername** [comment]
You can use GitHub hosted Actions runners to build on macOS without having to have a Mac yourself.
